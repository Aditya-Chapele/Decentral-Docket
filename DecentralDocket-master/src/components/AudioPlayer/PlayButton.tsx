/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import React from "react"
import { StyleSheet, TouchableOpacity } from "react-native"
import FontAwesome from "react-native-vector-icons/FontAwesome"

export default function PlayButton({
  state,
  onPress,
}: {
  state: "play" | "pause"
  onPress: () => void
}) {
  return (
    <TouchableOpacity style={styles.playButtonContainer} onPress={onPress}>
      <FontAwesome name={state} size={25} color="white" />
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  playButtonContainer: {
    // backgroundColor: '#000',
    borderColor: "rgba(93, 63, 106, 0.2)",
    // borderWidth: 16,
    width: 30,
    height: 30,
    borderRadius: 30,
    alignItems: "center",
    justifyContent: "center",
    // marginHorizontal: 32,
    shadowColor: "#5D3F6A",
    // shadowRadius: 30,
    shadowOpacity: 0.5,
  },
})
