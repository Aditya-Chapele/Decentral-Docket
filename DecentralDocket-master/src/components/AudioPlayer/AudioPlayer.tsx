/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import React, { useEffect, useState } from "react"
import {
  StyleSheet,
  Text,
  View,
  Image,
  SafeAreaView,
  TouchableOpacity,
} from "react-native"
import Slider from "react-native-slider"
import AudioRecorderPlayer from "react-native-audio-recorder-player"
import PlayButton from "./PlayButton"
import MaterialIcons from "react-native-vector-icons/MaterialIcons"

import { commonColors, textStyles } from "../../../docs/config"
import { heightPercentageToDP } from "react-native-responsive-screen"
const { primaryColor } = commonColors

export default function AudioPlayer({
  audioUrl,
  closePlayer,
}: {
  audioUrl: any
  closePlayer: () => void
}) {
  const [isAlreadyPlay, setisAlreadyPlay] = useState(false)
  const [duration, setDuration] = useState(0)
  const [timeElapsed, setTimeElapsed] = useState(0)
  const [percent, setPercent] = useState(0)
  const [audioRecorderPlayer, setAudioRecordedPlayer] = useState(
    new AudioRecorderPlayer()
  )

  const changeTime = async (seconds: any) => {
    // 50 / duration
    const seektime = (seconds / 100) * duration
    setTimeElapsed(seektime)
    audioRecorderPlayer.seekToPlayer(seektime)
  }
  const onStartPress = async () => {
    setisAlreadyPlay(true)
    audioRecorderPlayer.startPlayer(audioUrl)
    audioRecorderPlayer.setVolume(1.0)

    audioRecorderPlayer.addPlayBackListener(async (e) => {
      if (e.currentPosition === e.duration) {
        audioRecorderPlayer.stopPlayer()
        setisAlreadyPlay(false)
      }
      const currentPresent = Math.round(
        (Math.floor(e.currentPosition) / Math.floor(e.duration)) * 100
      )
      setTimeElapsed(e.currentPosition)
      setPercent(currentPresent)
      setDuration(e.duration)
    })
  }

  useEffect(() => {
    onStartPress()
    return () => {
      audioRecorderPlayer.stopPlayer()
      audioRecorderPlayer.removePlayBackListener()
    }
  }, [audioUrl])

  const onPausePress = async () => {
    setisAlreadyPlay(false)
    audioRecorderPlayer.pausePlayer()
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.buttonContainer}>
        {!isAlreadyPlay ? (
          <PlayButton onPress={() => onStartPress()} state="play" />
        ) : (
          <PlayButton onPress={() => onPausePress()} state="pause" />
        )}
      </View>
      <View
        style={{
          flexDirection: "row",
          alignItems: "center",
          justifyContent: "flex-start",
        }}
      >
        <View style={styles.seekbar}>
          <Slider
            minimumValue={0}
            maximumValue={100}
            trackStyle={styles.track}
            thumbStyle={styles.thumb}
            minimumTrackTintColor={commonColors.primaryDarkColor}
            onValueChange={(seconds: any) => changeTime(seconds)}
            value={percent}
          />

          <View style={styles.inprogress}>
            <Text style={[styles.textLight, styles.timeStamp]}>
              {audioRecorderPlayer.mmssss(Math.floor(timeElapsed))}
            </Text>
            <Text style={[styles.textLight, styles.timeStamp]}>
              {audioRecorderPlayer.mmssss(Math.floor(duration))}
            </Text>
          </View>
        </View>
        <TouchableOpacity onPress={closePlayer} style={{ marginLeft: "auto" }}>
          <MaterialIcons name="close" color={"white"} size={30} />
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  )
}

const styles = StyleSheet.create({
  container: {
    height: 50,
    backgroundColor: primaryColor,
    borderBottomLeftRadius: 5,
    borderBottomRightRadius: 5,
    justifyContent: "flex-start",
    alignItems: "center",
    flexDirection: "row",
    paddingHorizontal: 10,
    paddingBottom: 5,
  },
  textLight: {
    color: "rgba(255,255,255,1)",
  },
  text: {
    color: "#8E97A6",
  },
  textDark: {
    color: "rgba(255,255,255,1)",
  },
  buttonContainer: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
  },
  coverContainer: {
    shadowColor: "#5D3F6A",
    shadowOffset: { height: 15, width: 0 },
    shadowRadius: 8,
    shadowOpacity: 0.3,
  },

  track: {
    width: "100%",
    height: 2,
    borderRadius: 1,
    backgroundColor: "#FFF",
  },
  thumb: {
    width: 8,
    height: 8,
    backgroundColor: commonColors.primaryDarkColor,
  },
  timeStamp: {
    fontSize: 11,
    fontWeight: "500",
  },
  seekbar: { width: "80%", marginRight: 0, marginHorizontal: 16 },
  inprogress: {
    marginTop: -12,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  trackname: { alignItems: "center", marginTop: 32 },
})
