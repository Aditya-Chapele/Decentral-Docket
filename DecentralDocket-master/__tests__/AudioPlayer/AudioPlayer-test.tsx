import React from "react"
import renderer from "react-test-renderer"

import AudioPlayer from "../../src/components/AudioPlayer/AudioPlayer"

test("renders correctly", () => {
  const tree = renderer.create(<AudioPlayer audioUrl={"asdasd"} />).toJSON()
  expect(tree).toMatchSnapshot()
})
