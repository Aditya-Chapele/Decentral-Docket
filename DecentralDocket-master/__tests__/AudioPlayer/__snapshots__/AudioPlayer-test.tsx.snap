// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#003E9C",
      "borderBottomLeftRadius": 5,
      "borderBottomRightRadius": 5,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "flex-start",
      "paddingBottom": 5,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "rgba(93, 63, 106, 0.2)",
          "borderRadius": 30,
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#5D3F6A",
          "shadowOpacity": 0.5,
          "width": 30,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 25,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ÔÅã
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "margin": 32,
        "width": "80%",
      }
    }
  >
    <View
      animationType="timing"
      onLayout={[Function]}
      onValueChange={[Function]}
      step={0}
      style={
        [
          {
            "height": 40,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      thumbTouchSize={
        {
          "height": 40,
          "width": 40,
        }
      }
      value={0}
    >
      <View
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          [
            {
              "backgroundColor": "#b3b3b3",
            },
            {
              "borderRadius": 2,
              "height": 4,
            },
            {
              "backgroundColor": "#FFF",
              "borderRadius": 1,
              "height": 2,
              "width": "100%",
            },
          ]
        }
      />
      <View
        collapsable={false}
        renderToHardwareTextureAndroid={true}
        style={
          {
            "backgroundColor": "#2775EA",
            "borderRadius": 1,
            "height": 2,
            "opacity": 0,
            "position": "absolute",
            "width": 0,
          }
        }
      />
      <View
        collapsable={false}
        onLayout={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          {
            "backgroundColor": "#2775EA",
            "borderRadius": 10,
            "height": 8,
            "opacity": 0,
            "position": "absolute",
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": 0,
              },
            ],
            "width": 8,
          }
        }
      />
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        renderToHardwareTextureAndroid={true}
        style={
          [
            {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {},
          ]
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": -12,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "rgba(255,255,255,1)",
            },
            {
              "fontSize": 11,
              "fontWeight": "500",
            },
          ]
        }
      >
        00:00:00
      </Text>
      <Text
        style={
          [
            {
              "color": "rgba(255,255,255,1)",
            },
            {
              "fontSize": 11,
              "fontWeight": "500",
            },
          ]
        }
      >
        00:00:00
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
